src/test/java/com/thealgorithms/backtracking/ParenthesesGeneratorTest.java

```diff
@@ -0,0 +1,33 @@
+package com.thealgorithms.backtracking;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class ParenthesesGeneratorTest {
+    @ParameterizedTest
+    @MethodSource("tcStream")
+    void numberTests(int input, List<String> expected) {
```

Comment:
Sorry for a bad name suggestion before (copy-paste error). What do you think about:
```suggestion
    void regularInputTests(int input, List<String> expected) {
```

Reply:
Look good.